src/Raven.Server/Config/Categories/ConfigurationCategory.cs

```diff
@@ -204,9 +204,44 @@ public void Initialize(Func<string, SettingValue> getSetting, string serverDataD
                             else if (property.Info.PropertyType == typeof(HashSet<string>))
                             {
                                 var hashSet = new HashSet<string>(SplitValue(value), StringComparer.OrdinalIgnoreCase);
-
+                                
+                                property.Info.SetValue(this, hashSet);
+                            }
+                            else if (property.Info.PropertyType == typeof(HashSet<int>))
+                            {
+                                var values = SplitValue(value).Select(int.Parse).ToArray();
+                                HashSet<int> hashSet = new(values);
                                 property.Info.SetValue(this, hashSet);
                             }
+                            else if (property.Info.PropertyType.IsGenericType &&
+                                     property.Info.PropertyType.GenericTypeArguments[0].BaseType == typeof(Enum))
```

Comment:
What is this for? Does it add support for collection of enums?

Reply:
Yes, this is for HashSet<Enum>, in particular.